core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -578,8 +581,28 @@ private SnapshotInfo inProgressSnapshot(SnapshotsInProgress.Entry entry) {
                         shardSnapshotStatus.failure(shardFailure.reason());
                         shardStatus.put(shardId, shardSnapshotStatus);
                     } else {
-                        IndexShardSnapshotStatus shardSnapshotStatus =
-                            repository.getShardSnapshotStatus(snapshotInfo.snapshotId(), snapshotInfo.version(), indexId, shardId);
+                        IndexShardSnapshotStatus shardSnapshotStatus;
+                        if (snapshotInfo.state() == SnapshotState.FAILED) {
+                            // If the snapshot failed, but the shard's snapshot does
+                            // not have an exception, it means that partial snapshots
+                            // were disabled and in this case, the shard snapshot will
+                            // *not* have any metadata, so attempting to read the shard
+                            // snapshot status will throw an exception.  Instead, we create
+                            // a status for the shard to indicate that the shard snapshot
+                            // could not be taken due to partial being set to false.
+                            shardSnapshotStatus = new IndexShardSnapshotStatus();
+                            shardSnapshotStatus.updateStage(IndexShardSnapshotStatus.Stage.FAILURE);
+                            String msg = "snapshot not taken because partial snapshots are not " +
+                                         "configured for this snapshot and another shard caused " +
+                                         "the snapshot to fail";
```

Comment:
This will be repeated for every single good shard in the status and might hide the shards that actually caused the issue. Maybe we can say something short here like "skipped". If we want to improve readability, maybe we should improve the error message on the snapshot itself, so we don't have to repeat this for every shard.